src/main/java/de/danielbechler/diff/identity/IdentityStrategy.java

```diff
@@ -0,0 +1,20 @@
+package de.danielbechler.diff.identity;
+
+public interface IdentityStrategy {
+
+	/**
+	 *
+	 * @param _this never null
+	 * @param o
+	 * @return
+	 */
+	public boolean equals(final Object _this, final Object o);
+
+	/**
+	 *
+	 * @param _this never null
+	 * @return
+	 */
+	public int hashCode(final Object _this);
```

Comment:
I really don't think it's a good idea to let users change such a low-level detail as the hashCode. Since we don't wrap the items in any way, reading and writing them to collections still uses their original equals/hashCode implementations. Changing the IdentityStrategy only changes the way java-object-diff treats the objects when building the node graph.

hashCode is only used by some collection types to segment the items into different buckets, so only a subset needs to be compared via equals (ideally). We have no control over this behavior, unless we create proxy objects with our custom equals and hashCode implementations. And even then it wouldn't be guaranteed that only these proxies are used.

I can only think of one place where the hashCode actually matters for this library but I'll get to that later.

I suggest to remove hashCode from the interface for now and see how it goes. It's always easier to add new features than to change or remove existing ones, so let's see if this is something people actually need. We can always add it later.